src/commands/PlayCommand.ts

```diff
@@ -31,73 +31,122 @@ export class PlayCommand extends BaseCommand {
+                            if (response.first()?.content === "c" || response.first()?.content === "cancel") {
+                                return message.channel.send(createEmbed("warn", "The music selection has canceled"));
+                            }
+                            const videoIndex = parseInt(response.first()?.content as string);
+                            video = await this.client.youtube.getVideo(videos[videoIndex - 1].id);
+                        }
+                    } catch (err) {
+                        this.client.logger.error("YT_SEARCH_ERR:", err);
+                        return message.channel.send(createEmbed("error", `I could not obtain any search results\nError: **\`${err.message}\`**`));
+                    }
+                    return this.handleVideo(video, message, voiceChannel);
+                }
+                return this.handleVideo(video, message, voiceChannel);
             }
-        } else if (/^(?:https:\/\/open\.spotify\.com\/(?:user\/[A-Za-z0-9]+\/)?|spotify:)(album|playlist|track)(?:[/:])([A-Za-z0-9]+).*$/g.exec(url)) {
-            const input = /^(?:https:\/\/open\.spotify\.com\/(?:user\/[A-Za-z0-9]+\/)?|spotify:)(album|playlist|track)(?:[/:])([A-Za-z0-9]+).*$/g.exec(url) as any;
-            if (input[1] === "track") {
+        } else if (urlData.source === "spotify") {
+            if (urlData.type === "track") {
                 const trackData = await getData(url);
-                const trackSearchString = `${trackData.artists[0].name} - ${trackData.name}`;
+                const trackSearchString = `${trackData.artists[0].name} - ${trackData.name} Audio`;
                 const videoResult = await ytsr(trackSearchString, { limit: 1, safeSearch: false });
                 const queuedVideo = await this.client.youtube.getVideo((videoResult.items[0] as SRVideo).id);
                 return this.handleVideo(queuedVideo, message, voiceChannel);
-            } else if (input[1] === "playlist") {
+            } else if (urlData.type === "playlist") {
                 try {
                     const playlistData = await getData(url);
-                    const playlistSearchStrings: string[] = await playlistData.tracks.items.map((item: any): string => `${item.track.artists[0].name} - ${item.track.name}`);
+                    const playlistSearchStrings: string[] = await playlistData.tracks.items.map((item: any): string => `${item.track.artists[0].name} - ${item.track.name} Audio`);
                     this.client.logger.debug(playlistSearchStrings.join("\n"));
                     const addingPlaylistVideoMessage = await message.channel.send(
-                        createEmbed("info", `Adding all music in **[${playlistData.name}](${playlistData.external_urls.spotify})** playlist, please wait...`)
+                        createEmbed("info", `Adding all music in **[${playlistData.name}](${playlistData.external_urls.spotify})** playlist please wait...`)
```

Comment:
According to line 51
```suggestion
                        createEmbed("info", `Adding all music in **[${playlistData.name}](${playlistData.external_urls.spotify})** playlist, please wait...`)
```